engines/interp_seq_mc.cpp

```diff
@@ -172,10 +171,14 @@ bool InterpSeqMC::step(int i)
       // replay it in the solver with model generation
       concrete_cex_ = true;
     } else if (r.is_sat() && !bmc_res.is_sat()) {
-      throw PonoException("Internal error: Expecting satisfiable result");
+      if (bmc_res.is_unsat()) {
+        throw PonoException("expecting satisfiable result");
+      } else {
+        throw InternalSolverException("solver could not compute BMC query");
```

Comment:
Minor rewording suggestion: To me "compute" sounds like constructing the formula rather than solving it.

```suggestion
        throw InternalSolverException("solver could not solve BMC query");
```